### DISCUSSION TITLE
Raffle Deployment VRF Issue

### DISCUSSION THREAD

**Discussion by Muhdhammad** [Cyfrin/foundry-full-course-cu#2297] [General]
Hey everyone,

I followed the raffle lesson and deployed the contract to the Sepolia testnet. My contract was verified automatically, so I didn’t need to use the ```.json``` method. I funded my VRF subscription with 15 LINK and the automation with 3 LINK. However, the transaction for the VRF has failed with the error: "Invalid Key Hash." Now, when I try to call ```enterRaffle``` again, it's not allowing me because the ```RaffleState``` is stuck at calculating (1). Any ideas on how to resolve this? Here are the details:

Contract Address: 
https://sepolia.etherscan.io/address/0x4ba6DF61611AD963ad0Fc2295D00F8fc03bC6460

Failed Transaction Hash of VRF Subscription:
https://sepolia.etherscan.io/tx/0x27743380b37a2e08474e1ab454e8fbf7f158ba78fd8c3503090363f206d4e42e

![e2](https://github.com/user-attachments/assets/1d5faff9-83db-416c-b926-5828b7bc4fb2)

**Comment by EngrPips** [comment]
Hello @Muhdhammad, I will advise you to deploy a new lottery contract and use the valid `key hash` this time.

**Reply by Muhdhammad** [reply]
valid ```key hash``` ?

**Reply by EngrPips** [reply]
You don't know what a `key hash` is? It is what you set in your lottery contract constructor.
```js
 i_gasLane = _gasLane;
``` 
And you use it when requesting a random number from the `VRFCoordinator` contract. 

```js
 uint256 requestId = i_vrfCoordinatorContract.requestRandomWords(VRFV2PlusClient.RandomWordsRequest({
            keyHash: i_gasLane, <@ here
            subId: i_subscriptionId,
            requestConfirmations: REQUEST_CONFIRMATIONS,
            callbackGasLimit: CALLBACK_GAS_LIMIT,
            numWords: NUM_OF_WORDS,
            // Set nativePayment to true to pay for VRF requests with Sepolia ETH instead of LINK
            extraArgs: VRFV2PlusClient._argsToBytes(VRFV2PlusClient.ExtraArgsV1({nativePayment: false}))
        }));
```
